011-admin-server.md

```diff
@@ -0,0 +1,139 @@
+The implementation of the APIs that the server then exposes can be created independently and will be automatically included in the running server if they are available to the Service Loader.
+
+The reason for the modular approach is that it allows the strimzi project to create a base set of functionality that can easily be extended by a downstream project.
+Not only is it easy to extend the API but it can be done in isolation.
+This removes the requirement of the extensions needing to track the strimzi source repositories.
+For example, lets assume that strimzi only implements a Kafka administration API which provides a set of APIs based around the Kafka Admin Client.
+A downstream project then would like to add a Schema Registry to the cluster and needs to add an API in support of the new functionality.
+With the modular server, they can define the API and its implementation, bundle the new functionality into a JAR file and add the new JAR file to the classpath.
+A secondary benefit of a modular approach is that it allows the strimzi user to configure the server to only contain the functionality that they would like to use providing for a leaner server.
+
+The classpath of the server can be configured at build time or run time.
+To configure at build time, the server and dependant modules can be bundled in a fat-JAR and obviously, at run time, the JVM can be configured in multiple ways to include JAR files on the classpath.
+
+### REST API Implementations
+REST modules would use the Vert.x Web API Contract support and specifically the OpenAPI3 support to define the shape of the API and bind the operations to a set of handlers.
+The advantage of using the OpenAPI3 definition is the support it contains for validation which allows you to build a more secure API without having to pollute the implementation with validation code.
+It leaves the implementation of the operations to focus entirely on the business logic.
+A secondary advantage of using the OpenAPI3 specification is the numerous tools that are freely available to create generated documentation of the API.
+ 
+The Vert.x OpenAPI3 support will read an OpenAPI3 specification YAML file and allow the developer to map the original operations to handler methods.
+It creates a Vert.x Router which can be mounted on a Vert.x HTTP/S server.
+The module would implement the service interface defined by the server which will contain a method that the server can call to load the OpenAPI3 router and mount the router as a subrouter at a specified mount point to the root path.
+So, for example, if the OpenAPI3 spec defined endpoints `GET /topics` and `POST /topics` and these were mounted on the root path at the mount point `/admin` then the server would listen for incoming requests of `GET /admin/topics` and `POST /admin/topics` and route them to the handlers defined in the module.
```

Comment:
So with reference to my previous comment, I'm not sure if any plugin should be bindable to the root path. The root path could be used for discovery of the API provided.

Reply:
So if the response in the previous comment is sufficient.